docs/library/jesd204/ad_ip_jesd204_tpl_adc/index.rst

```diff
@@ -0,0 +1,335 @@
+   :no-type-info:
+
+Theory of Operation
+--------------------------------------------------------------------------------
+
+Interfaces and Signals
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Configuration Interface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The peripheral features a register map configuration interface that can be
+accessed through the AXI4-Lite ``S_AXI`` port. The register map can be used to
+configure the peripherals operational parameters, query the current status of
+the device and query the features supported by the device.
+
+Link layer interface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The link layer interface description can be found in the 
+:ref:`User Data Interface <axi_jesd204_rx_user_data>` section of the 
+:ref:`JESD204B/C Link Receive Peripheral <axi_jesd204_rx>` IP.
+
+Application layer interface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The application layer is connected to the deframer block output. The deframer
+module creates sample data from the lane mapped and formatted JESD204 link data
+based on the specified deframer configuration.
+
+The data in the application layer interface ``adc_data`` has the following
+layout:
+
+::
+
+       MSB                                                               LSB
+     [ MmSn, ..., MmS1, MnS0, ..., M1Sn, ... M1S1, M1S0, M0Sn, ... M0S1, M0S0 ] 
+
+Where MjSi refers to the i-th sample of the j-th converter. With m being the
+number of converters and n the number of samples per converter per beat.
+
+The core asserts the ``enable`` signal for each channel that is enabled by the
+software.
+
+Clock Monitor
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``REG_STATUS`` (``0x054``) register ``CLK_FREQ`` field allows to determine
+the clock rate of the device clock (``link_clk``) relative to the AXI interface
+clock (``s_axi_aclk``). This can be used to verify that the device clock is
+running at the expected rate.
+
+The number is represented as unsigned 16.16 format. Assuming a 100MHz processor
+clock this corresponds to a resolution of 1.523kHz per LSB. A raw value of 0
+indicates that the link clock is currently not active.
+
+Data Formatter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The component is configured by the ``REG_CHAN_CNTRL`` register
+``FORMAT_SIGNEXT,FORMAT_TYPE,FORMAT_ENABLE`` fields. The block introduces one
+clock cycle latency.
+
+PRBS Check
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The block can monitor and compare the incoming deframed raw data against
+PN9,PN23 and PN7, PN15 (if enabled) patterns selected by the ``ADC_PN_SEL``
+field of ``REG_CHAN_CNTRL_3`` register.
+
+.. 
+  it is missing PN31, because it does not exist on the source code
```

Comment:
it is there because this signal exists in the wiki document, but it is not available inside component.xml. It is supposed not to render, it is just a warning for the people reviewing this doc.